src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -172,5 +165,190 @@ private static JToken ParseToJToken(string value)
+                History = historyArray,
+            };
+        }
+
+        private static void UpdateProperties(ref JToken jsonToken, List<string> propertyNames, bool hideInput = false, bool hideResult = false)
+        {
+            if (jsonToken == null)
+            {
+                return;
+            }
+
+            if ((hideInput || hideResult) && propertyNames == null)
+            {
+                propertyNames = new List<string>();
+            }
+
+            if (hideInput)
+            {
+                propertyNames.Add("Input");
+            }
+
+            if (hideResult)
+            {
+                propertyNames.Add("Result");
+            }
+
+            foreach (var propertyName in propertyNames)
+            {
+                UpdateProperties(ref jsonToken, propertyName);
+            }
+        }
+
+        private static void UpdateProperties(ref JToken jsonToken, string properyName, bool hideInput = false, bool hideResult = false)
+        {
+            if (jsonToken == null)
+            {
+                return;
+            }
+
+            if (hideInput)
+            {
+                jsonToken["Input"]?.Parent?.Remove();
+            }
+
+            if (hideResult)
+            {
+                jsonToken["Result"]?.Parent?.Remove();
+            }
+
+            if (!string.IsNullOrEmpty(properyName))
+            {
+                jsonToken[properyName]?.Parent?.Remove();
+            }
+        }
+
+        private static void TrackNameAndScheduledTime(JToken historyItem, EventType eventType, int index, ref Dictionary<string, EventIndexDateMapping> eventMapper)
+        {
+            eventMapper.Add($"{eventType}_{historyItem["EventId"]}", new EventIndexDateMapping { Index = index, Name = historyItem["Name"].ToString(), Input = historyItem["Input"]?.ToString(), Date = (DateTime)historyItem["Timestamp"] });
+        }
+
+        private static void AddScheduledEventDataAndAggreagate(IReadOnlyDictionary<string, EventIndexDateMapping> eventMapper, string prefix, JToken historyItem, ref List<int> indexList)
+        {
+            var taskScheduledData = eventMapper[$"{prefix}_{historyItem["TaskScheduledId"]}"];
+            historyItem["ScheduledTime"] = taskScheduledData.Date;
+            historyItem["FunctionName"] = taskScheduledData.Name;
+            indexList.Add(taskScheduledData.Index);
+        }
+
+        private static void ConvertOutputToJToken(ref JToken jsonToken, bool showHistoryOutput)
+        {
+            if (!showHistoryOutput)
+            {
+                return;
+            }
+
+            if (jsonToken == null)
+            {
+                return;
+            }
+
+            jsonToken["Result"] = ParseToJToken(jsonToken["Result"]?.ToString());
+        }
+
+        private static void ChangePrpertyName(ref JToken jsonToken, string currentPropertyName, string newPropertyName)
+        {
+            var value = jsonToken?[currentPropertyName];
```

Comment:
The use of `jsonToken?` null-checks seems pretty excessive. I suggest you should leave these null-checks out. If you think null is a possibility for a history item, then I suggest you filter these out in the calling code (e.g. at the top of the `for` loop) so that these helper methods are never called with null inputs.

Reply:
Great point. I  decided to remove this method since it is called only once and with the use of `JObject` becames trivial. Thank you!